JASP-Engine/JASP/R/correlation.R

```diff
@@ -15,806 +15,721 @@
+            next
+          } else if (identical(pair, "")) { # above the diagonal
+            row[[col.est]] <- row[[col.p]] <- row[[col.mpr]] <- row[[col.upperCI]] <- row[[col.lowerCI]] <- ""
+            next
+          }
+          
+        }
+        
+        pairName <- paste(sort(pair), collapse="-")
+        result <- tableData$result[[pairName]][[test]]
+        
+        if (flagSignificant && is.numeric(result$p.value)) {
+
+          if (result$p.value < .001) {
+            row.footnotes[[col.est]] <- list("***")
+          } else if (result$p.value < .01) {
+            row.footnotes[[col.est]] <- list("**")
+          } else if (result$p.value < .05) {
+            row.footnotes[[col.est]] <- list("*")
+          }
+          
+        }
+        
+        row[[col.est]] <- result$estimate
+        
+        if (reportSignificance)
+          row[[col.p]] <- result$p.value
+
+        if (reportVovkSellkeMPR)
+          row[[col.mpr]] <- result$MPR
+
+        if (reportCI) {
+          row[[col.upperCI]] <- result$upperCI
+          row[[col.lowerCI]] <- result$lowerCI
+        }
+        
+        if (! is.null(result$errorMessage)) {
+          index <- .addFootnote(footnotes, result$errorMessage)
+          row.footnotes[[col.est]] <- c(row.footnotes[[col.est]], list(index))
+        }
+        
+      } # end loop over columns
+
+    } # end loop over tests
+    
+    if (displayPairwise) {
+		  row[[".variable1"]] <- pair[1]
+      row[[".separator"]] <- "-"
+      row[[".variable2"]] <- pair[2]
+    } else {
+      row[[".variable"]] <- variables[i]
+    }
 
-				p.values <- list()
-				MPRs <- list()
-				upperCIs <- list()
-				lowerCIs <- list()
+		if (length(row.footnotes) > 0)
+			row[[".footnotes"]] <- row.footnotes
 
-				if (length(tests) > 1 || reportSignificance || (test == "pearson" && options$confidenceIntervals))
-					row[[length(row)+1]] <- test.names[[test]]
+		rows[[i]] <- row
+    
+	} # end loop over rows
 
-				if (reportSignificance)
-					p.values[[length(p.values)+1]] <- "p-value"
+	correlation.table[["data"]] <- rows
+	correlation.table[["footnotes"]] <- as.list(footnotes)
 
-				if (reportVovkSellkeMPR){
-					if (flagSignificant){
-						MPRs[[length(MPRs)+1]] <- "VS-MPR\u207A"
-					} else {
-						MPRs[[length(MPRs)+1]] <- "VS-MPR\u002A"
-					}
-				}
+	return(correlation.table)
+}
 
-				if (test == "pearson" && options$confidenceIntervals) {
+.plotCorrelations <- function(dataset, perform, options) {
 
-					upperCIs[[length(upperCIs)+1]] <- paste("Upper ", 100 * options$confidenceIntervalsInterval, "% CI", sep="")
-					lowerCIs[[length(lowerCIs)+1]] <- paste("Lower ", 100 * options$confidenceIntervalsInterval, "% CI", sep="")
-				}
+  variables <- unlist(options$variables)
+  correlation.plot <- NULL
 
-				for (j in .seqx(1, i-1)) {
+   if (perform == "init" && length(variables) > 1) {
+     
+    l <- length(variables)
```

Comment:
numberOfVars

Reply:
Pretty much the only change in the plotting function was moving it from point A (main analysis function) to B (its own sub-function). It probably needs a bigger rework later.